src/internal/types/themeData.ts

```diff
@@ -0,0 +1,15 @@
+export type ThemeSaveState = {
+  history: any; // json object
+  index: number;
+  hash: string;
+};
+
+export type ThemeHistories = {
+  histories: History[];
+  index?: number;
+};
+
+export type History<D = any> = {
+  state: D;
```

Comment:
What do other people think about this property name? Obviously it aligns with Puck but we don't need to. Maybe `value`?